samcli/lib/build/app_builder.py

```diff
@@ -127,6 +123,49 @@ def build(self):
 
         return result
 
+    def _get_build_graph(self):
+        """
+        Converts list of functions into a build graph, where we can iterate on each unique build and trigger build
+        :return: BuildGraph, which represents list of unique build definitions
+        """
+        build_graph = BuildGraph(self._build_dir)
+        functions = self._resources_to_build.functions
```

Comment:
do we account for building layers too? Should we?

Reply:
No we don't, and we shouldn't. Because functions may have multiple function definition in the same codeuri and runtime, because of the entry-point (you can write multiple functions and keep them in the same package, but your entry-point should be different). But for layer's, I couldn't think of any reason, why you will crate 2 different layers out of same codeuri and runtime, where you could just create one layer and use it.